lib/wmic.js

```diff
@@ -1,106 +1,118 @@
-    curr = pid[i]
-    if (!options.history[curr]) options.history[curr] = {}
-    whereClause += ' or ' + prefix + curr
+/**
+  * Get pid informations through wmic command.
+  * @param  {Number[]} pids
+  * @param  {Object} options
+  * @param  {Function} done(err, stat)
+  */
+function wmic (pids, options, done) {
+  var whereClause = 'ProcessId=' + pids[0]
+  for (var i = 1; i < pids.length; i++) {
+    whereClause += ' or ' + 'ProcessId=' + pids[i]
   }
 
-  // http://social.msdn.microsoft.com/Forums/en-US/469ec6b7-4727-4773-9dc7-6e3de40e87b8/cpu-usage-in-for-each-active-process-how-is-this-best-determined-and-implemented-in-an?forum=csharplanguage
-  var args = ['PROCESS', 'where', '"' + whereClause + '"', 'get', 'ProcessId,workingsetsize,usermodetime,kernelmodetime']
-  var wmic = spawn('wmic', args, {detached: true, windowsHide: true, windowsVerbatimArguments: true})
-  var stdout = ''
-  var stderr = ''
-  var error = ''
-
-  // Note: On Windows the returned value includes fractions of a second. Use Math.floor() to get whole seconds.
-  var uptime = Math.floor(os.uptime())
-
-  wmic.stdout.on('data', function (d) {
-    stdout += d.toString()
-  })
-
-  wmic.stderr.on('data', function (d) {
-    stderr += d.toString()
-  })
-
-  wmic.on('error', function (err) {
-    error = '[pidusage] Command "wmic ' + args.join(' ') + '" failed with error ' + err.message
-  })
+  var args = [
+    'PROCESS',
+    'where',
+    '"' + whereClause + '"',
+    'get',
+    'CreationDate,KernelModeTime,ParentProcessId,ProcessId,UserModeTime,WorkingSetSize'
+  ]
 
-  wmic.on('close', function (code) {
+  bin('wmic', args, {windowsHide: true, windowsVerbatimArguments: true}, function (err, stdout) {
+    if (err) return done(err)
     var date = Date.now()
-    stdout = stdout.trim()
-    stderr = stderr.trim()
 
-    if (!stdout || code !== 0) {
-      error += format('%s %s Wmic errored, please open an issue on https://github.com/soyuka/pidusage with this message.%s', os.EOL, new Date().toString(), os.EOL)
-      error += format('Command was "wmic %s" %s System informations: %s - release: %s %s - type %s %s', args.join(' '), os.EOL, os.EOL, os.release(), os.EOL, os.type(), os.EOL)
-      stderr = error + (stderr ? format('Wmic reported the following error: %s.', stderr) : 'Wmic reported no errors (stderr empty).')
-      stderr = format('%s%s%sWmic exited with code %d.', os.EOL, stderr, os.EOL, code)
-      stderr = format('%s%sStdout was %s', stderr, os.EOL, stdout || 'empty')
+    // Note: On Windows the returned value includes fractions of a second.
+    // Use Math.floor() to get whole seconds.
+    var uptime = Math.floor(os.uptime())
 
-      return done(new Error(stderr), null)
-    }
+    // Example of stdout on Windows 10
+    // CreationDate: is in the format yyyymmddHHMMSS.mmmmmmsUUU
+    // KernelModeTime: is in units of 100 ns
+    // UserModeTime: is in units of 100 ns
+    // WorkingSetSize: is in bytes
+    //
```

Comment:
Please add the information that results are coming back in their keys alphabetical order (useful if we add one to the future even though you did good at settings them at that order already above).

Reply:
done